backend/src/routers/user_auth/login.py

```diff
@@ -0,0 +1,62 @@
+from datetime import timedelta
+from fastapi import Depends, HTTPException, APIRouter
+from fastapi.encoders import jsonable_encoder
+from fastapi.security import OAuth2PasswordRequestForm
+from starlette.status import HTTP_401_UNAUTHORIZED
+from starlette.responses import Response
+from .utils import (
+    get_user,
+    verify_password,
+    MOCK_DB,
+    REFRESH_TOKEN_EXPIRE_MINUTES,
+    ACCES_TOKEN_EXPIRES_MINUTES,
+    create_token,
+)
+
+
+router = APIRouter()
+
+
+def authenticate_user(fake_db, username: str, password: str):
+    user = get_user(fake_db, username)
+    if not user:
+        return False
+    if not verify_password(password, user.hashed_password):
+        return False
+    return user
+
+
+@router.post("/login")
+async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends()):
+    user = authenticate_user(MOCK_DB, form_data.username, form_data.password)
+    if not user:
+        print("User NOT verified.")
```

Comment:
Instead of `print`s, consider using `logging` package so that you can control the logging output between production and development (I know it is really not necessary for you since this isn't a very serious project, but it's good to know for the future). Also add a little more information to the log so it is more useful, like the username or the reason why it failed.

Reply:
Actually, uvicorn gives nicely described logs by itself (Im sure they come from package `logging` you mentioned); I'll keep it in mind and try to improve logging in future.